### ISSUE TITLE
[Bug]: Virtual files are not deleted if files/folders are deleted in the server

### ISSUE THREAD

**Issue by wonx** [nextcloud/desktop#4384]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

I deleted a folder containing many files in the server. That folder is an external storage (sftp), and nextcloud was synced usin occ scan. Nextcloud recognised the changes and deleted those files from the database, and everything showed as expected in the web interface.

However, the two computers I use (with virtual files), never deleted that folder and files.

The nextcloud client just ignores those changes and let the virtual files stay where they were. I can see that it "sees" something as for a moment I can see the deleted folder appear in the progress bar, but just for a few seconds and then the client shows the green check mark.

It happens both with the Windows and Linux client, in two different computers.

If I try to open one of these files, nothing happens in the client, whereas I see a http 404 error in the server log.

### Steps to reproduce

Delete folder containing many (tens of thousands) of files in the server. Wait until the client sees these changes.

### Expected behavior

The virtual files corresponding to the deleted files in the server should also be deleted in the clients.

### Which files are affected by this bug

.

### Operating system

Windows

### Which version of the operating system you are running.

Windows 10 and Ubuntu 20.04

### Package

Other

### Nextcloud Server version

23.0.2

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by wonx** [comment]
More info and screenshots in #4253

**Comment by jdinkel** [comment]
I'm experiencing this same behavior.  Deleted files from the server are not removed from the virtual files on the Windows clients.  If I try to open one of the files, I get errors about unable to read the file but it still doesn't go away.  I've noticed it does go away after a reboot, but to be usable I would expect it to go away automatically within a few seconds.

I've used Nextcloud for years but just did a big reorganization of my files and my workflow based on using virtual files rather than synchronized files/folders.
